packages/types/src/codec/EnumType.ts

```diff
@@ -98,15 +98,19 @@ export default class EnumType<T> extends Base<Codec> implements Codec {
     const aliasKey = aliasses[keyLower] || keyLower;
     const index = keys.indexOf(aliasKey);
 
-    assert(index !== -1, `Cannot map input on JSON, unable to find '${key}' in ${keys.join(', ')}`);
+    assert(index !== -1, `Cannot map EnumType JSON, unable to find '${key}' in ${keys.join(', ')}`);
 
     return EnumType.createValue(def, index, value);
   }
 
   private static createValue (def: TypesDef, index: number = 0, value?: any): Decoded {
+    const Clazz = Object.values(def)[index];
+
+    assert(!isUndefined(Clazz), `Unable to create EnumType via index ${index}, in ${Object.keys(def).join(', ')}`);
```

Comment:
Previously we just failed with invalid errors, now we do an explicit check that there is an value - more info on parsing failures